src/app/api/categories/[...handle]/route.ts

```diff
@@ -0,0 +1,179 @@
+    return notFound()
+  }
+
+  const {
+    rows: products,
+    metadata: { count },
+  } = await getProductsByCategoryId(category.id, searchParams)
+
+  const publishedProducts: ProductDTO[] = products.filter(
+    (product) => product.status === "published"
+  )
+
+  const nextPage = parseInt(page) + parseInt(limit)
+
+  return NextResponse.json({
+    product_categories: Object.values(product_categories),
+    response: {
+      products: publishedProducts,
+      count,
+    },
+    nextPage: count > nextPage ? nextPage : null,
+  })
+}
+
+async function getProductsByCategoryId(
+  category_id: string,
+  params: Record<string, any>
+): Promise<{ rows: ProductDTO[]; metadata: Record<string, any> }> {
+  // Extract the query parameters
+  let { currency_code } = params
+
+  currency_code = currency_code && currency_code.toUpperCase()
+
+  const { query, modules } = await MedusaApp({
+    modulesConfig: [
+      {
+        module: Modules.PRODUCT,
+        path: "@medusajs/product",
+      },
+      {
+        module: Modules.PRICING,
+        path: "@medusajs/pricing",
+      },
+    ],
+    sharedResourcesConfig: {
+      database: { clientUrl: process.env.POSTGRES_URL },
+    },
+  })
+
+  const filters = {
+    take: parseInt(params.limit) || 100,
+    skip: parseInt(params.offset) || 0,
+    filters: {
+      category_id: [category_id],
+    },
+    currency_code,
+  }
+
+  const productsQuery = `#graphql
+    query($filters: Record, $take: Int, $skip: Int) {
+      products(filters: $filters, take: $take, skip: $skip) {
+        id
+        title
+        handle
+        tags
+        status
+        collection
+        collection_id
+        thumbnail
+        images {
+            url
+            alt_text
+            id
+        }
+        options {
+            id
+            value
+            title
+        }
+        variants {
+            id
+            title
+            created_at
+            updated_at
+            thumbnail
+            inventory_quantity
+            material
+            weight
+            length
+            height
+            width
+            options {
+                id
+                value
+                title
+            }
+            price {
+              price_set {
+                id
+              }
+          }
+        }
+      }
+    }`
+
+  const { rows, metadata } = await query(productsQuery, filters)
+
+  for (const product of rows) {
+    for (const variant of product.variants) {
+      const priceSetIds = variant.price.map((price) => price.price_set.id)
+
+      const prices = await modules.pricingService.calculatePrices(
+        { id: priceSetIds },
+        {
+          context: { currency_code },
+        }
+      )
+
+      const price = prices.find(
+        (price) => price.currency_code === currency_code
+      )
+
+      if (!price) continue
+
+      delete variant.price
+      variant.price = price
+      variant.calculated_price = price.amount
+    }
+  }
```

Comment:
```suggestion
      const [price] = await modules.pricingService.calculatePrices(
        { id: priceSetIds },
        {
          context: { currency_code },
        }
      )

      delete variant.price
      
      if (!price) continue

      variant.price = price
      variant.calculated_price = price.amount
    }
  }
```

Reply:
This doesn't work as expected if the product has multiple priceSetIds. [price] on line 156 will return the first array item, but that's not necessarily the one that matches the currency code. That's why I added the .find, to grab the price that matches the requested currency code. Or am I missing something here?

Reply:
pricingService.calculatePrices is throwing a TS error, might be related:

```
Property 'calculatePrices' does not exist on type '{ __joinerConfig: ModuleJoinerConfig; __definition: ModuleDefinition; } | { __joinerConfig: ModuleJoinerConfig; __definition: ModuleDefinition; }[]'.
  Property 'calculatePrices' does not exist on type '{ __joinerConfig: ModuleJoinerConfig; __definition: ModuleDefinition; }'.
```